ql/src/java/org/apache/hadoop/hive/ql/parse/GenTezUtils.java

```diff
@@ -76,6 +80,24 @@ public static UnionWork createUnionWork(
     tezWork.add(unionWork);
     return unionWork;
   }
+  
+  private static boolean isRestrictReducerExtrapolation(GenTezProcContext context) {
+
+    Table table = context.parseContext.getContext().getLoadTableOutputMap().values()
+            .stream().findFirst().get().getTable();
+
+    // Iceberg table
+    if (DDLUtils.isIcebergTable(table)) {
+      return true;
+    }
+
+    // Hive ACID table
+    if (!table.isNonNative() && AcidUtils.isTransactionalTable(table)) {
```

Comment:
I am not sure it's required for ACID tables as we have built-in compaction support

Reply:
Done, removed.